src/charm.py

```diff
@@ -217,16 +229,9 @@ def _on_mongod_pebble_ready(self, event) -> None:
             event.defer()
             return
 
-        if not self.get_secret("app", "keyfile"):
```

Comment:
Shouldn't we keep this check? and defer the event until the secrets are initialized in `_relation_changes_handler`

Reply:
No, we don't because the secrets will be created in `_relation_changes_handler` in LeaderElected event which happens before `on_mongod_pebble_ready`